datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -29,7 +40,15 @@
 /**
  * Performs partition assignment based on partition throughput information
  */
-public class LoadBasedPartitionAssigner {
+public class LoadBasedPartitionAssigner implements MetricsAware {
+  private static final Logger LOG = LoggerFactory.getLogger(LoadBasedPartitionAssignmentStrategy.class.getName());
+  private static final String CLASS_NAME = LoadBasedPartitionAssignmentStrategy.class.getSimpleName();
```

Comment:
I'm not sure if you should use a different strategy class name here. This removes the flexibiilty of reuse of this class in the new strategy. This same thing applies to the LOGGER as well. Since the changes are in open-source, there is always a possibility of new strategy.

Reply:
Let's worry about that when we have another strategy that uses `LoadBasedPartitionAssigner`. At the moment it's only used by `LoadBasedPartitionAssignmentStrategy`.

Reply:
Both of these cases are easily achieved by tags in metrics while graphing them, but I don't think ingraphs have tags as of now. But for this one, I would also prefer to keep the class name as it is.

Reply:
@jzakaryan I thought about this use-case again and I could think of bunch of alternatives:

a) Pass Logger and Metric Prefix in the class constructor and use it. There are other instances in this repo where Logger is passed instead of created. I can think of this to be the main reason.

b) If you want the stronger stitching, then you can move this class as inner class of the strategy. Based on our offline discussion, you have less preference for this.

So, option (a) will unify both of our thoughts on this.